File: myddd-ioc/myddd-ioc-spring/src/test/java/org/myddd/ioc/spring/AbstractTest.java
package org.myddd.ioc.spring;


import org.junit.jupiter.api.BeforeEach;
import org.myddd.domain.InstanceFactory;
import org.myddd.domain.InstanceProvider;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.ComponentScan;

@SpringBootTest(classes = AbstractTest.class)
@SpringBootApplication
@ComponentScan(basePackages = {"org.myddd"})
@EntityScan(basePackages = {"org.myddd"})
public abstract class AbstractTest {

    @Autowired
    protected ApplicationContext applicationContext;

    protected InstanceProvider instanceProvider;

    @BeforeEach
    protected void beforeClass(){
        instanceProvider = SpringInstanceProvider.createInstance(applicationContext);
        InstanceFactory.setInstanceProvider(instanceProvider);
    }

}

File: examples/kotlin-ddd/infra/src/test/kotlin/org/myddd/kotlin/AbstractTest.kt
package org.myddd.kotlin

import org.junit.jupiter.api.BeforeEach
import org.myddd.domain.InstanceFactory
import org.myddd.ioc.spring.SpringInstanceProvider
import org.myddd.kotlin.domain.User
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.domain.EntityScan
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.ApplicationContext
import org.springframework.context.annotation.ComponentScan
import java.util.*
import jakarta.inject.Inject

@SpringBootTest(classes = [AbstractTest::class])
@SpringBootApplication
@ComponentScan(basePackages = ["tech.taoofcoding","org.myddd"])
@EntityScan(basePackages = ["tech.taoofcoding","org.myddd"])
abstract class AbstractTest {

    @Inject
    lateinit var applicationContext:ApplicationContext

    @BeforeEach
    fun beforeEach(){
        InstanceFactory.setInstanceProvider(SpringInstanceProvider.createInstance(applicationContext))
    }

    companion object {
        fun randomString():String {
            return UUID.randomUUID().toString().replace("-","")
        }
    }

    fun randomUser():User {
        val user = User()
        user.userId = randomString()
        user.name = randomString()
        user.phone = "18620501006"
        user.email = "${randomString()}@taoofcoding.tech"
        return user
    }
}